bodo/pandas/utils.py

```diff
@@ -817,9 +828,36 @@ def fallback_wrapper(attr):
     ):
 
         def silenced_method(*args, **kwargs):
+            msg = ""
             with warnings.catch_warnings():
                 warnings.simplefilter("ignore", category=BodoLibFallbackWarning)
-                return attr(*args, **kwargs)
+                try:
+                    return attr(*args, **kwargs)
+                except TypeError as e:
+                    msg = e
+                    pass
+
+            # In some cases, fallback fails and raises TypeError due to some operations being unsupported between PyArrow types.
+            # Below logic processes deeper fallback that converts problematic PyArrow types to their Pandas equivalents.
+            if isinstance(self, bodo.pandas.BodoSeries):
+                pd_self = pd.Series(self)
+
+                # When self.dtype is pd.ArrowDtype(pa.timestamp("ns")), apply to_datetime elementwise.
+                if isinstance(self.dtype, pd.ArrowDtype) and pa.types.is_timestamp(
```

Comment:
Needs to check for timestamps with time zones since `datetime64` cast below wouldn't be valid.